rapidpro/models/actions.py

```diff
@@ -53,12 +57,15 @@ def render(self):
 
 
 class SetContactFieldAction(Action):
-    def __init__(self, field_key, field_name, value):
+    def __init__(self, field_name, value):
         super().__init__('set_contact_field')
-        self.field_key = field_key
+        self.field_key = self._get_field_key(field_name)
```

Comment:
Shouldn't this have both options? If a key is provided then that is used otherwise the fieldname is used by default. In either case it would need to be checked/edited to conform to the RapidPro requirements for keys.